### PR TITLE
Merge configuration files into pyproject.toml

### PR DESCRIPTION
**Pull request by marcoesters**
### Description

Convert config files into `pyproject.toml` to use more modern python packaging practices.

The lower bounds for `setuptools` and `setuptools_scm[toml]` were taken from `menuinst`. I'm not sure if they need to be adjusted or can be removed altogether.

Also add setuptools to host dependencies since it will be removed from pip's run dependencies on conda-forge eventually.

### Checklist - did you ...

- [X] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
- [X] Add / update necessary tests?
- [X] Add / update outdated documentation?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,44 @@
+[build-system]
+requires = ["setuptools>=45", "setuptools_scm[toml]>=6.2"]
```
Comment (jaimergp) [inline pyproject.toml:2]: I'd say we need 
```suggestion
requires = ["setuptools>=70.1", "setuptools_scm[toml]>=6.2"]
```

To make sure `wheel` is around. xref https://github.com/conda-forge/pip-feedstock/issues/124#issuecomment-2261091498

File: .flake8
Comment (jaimergp) [inline .flake8:1]: Cannot we have this in pyproject.toml too?

File: constructor/_version.py
Comment (jaimergp) [inline constructor/_version.py:1]: We might need to add this path to .gitignore.

File: constructor/__init__.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
-from . import _version
-
-__version__ = _version.get_versions()['version']
+try:
+    from . import _version
+    __version__ = _version.__version__
+except ImportError:
+    __version__ = "dev"
```
Comment (jaimergp) [inline constructor/__init__.py:5]: I like this [idiom](https://github.com/conda/conda-libmamba-solver/blob/c25621abd4e4b0d9726223c617eaef5fcfa22807/conda_libmamba_solver/__init__.py#L4-L13) better:

```suggestion
try:
    from ._version import version as __version__
except ImportError:
    try:
        from importlib.metadata import version

        __version__ = version("constructor")
        del version
    except ImportError:
        __version__ = "0.0.0.unknown"
```

File: .flake8
Comment (marcoesters) [reply .flake8:1]: `flake8` unfortunately doesn't support `pyproject.toml`

**Review by jaimergp — APPROVED** [review]

File: .flake8
Comment (jaimergp) [reply .flake8:1]: Got it. We could move to `ruff` in a subsequent PR and get rid of this too.

**Pull request merged**